src/main/java/ifsul/agileproject/rachadinha/controller/UserController.java

```diff
@@ -20,57 +22,58 @@ public UserController(UserService userService) {
 
     //Buscar user por ID
     @GetMapping("{id}")
-    public User getUserByID(@PathVariable Integer id){
-        return userService
-                .findUserByID(id)
-                .orElseThrow(() ->
-                        new ResponseStatusException(HttpStatus.NOT_FOUND, "Usuário não encontrado.")
-                );
+    public ResponseEntity<UserRespostaDTO> getUserByID(@PathVariable Integer id){
+      User usuario = userService.findUserByID(id);
+      return new ResponseEntity<>(UserRespostaDTO.transformaEmDTO(usuario), HttpStatus.OK);
     }
 
     //Cadastrar user
     @PostMapping("/cadastro")
-    @ResponseStatus(HttpStatus.CREATED)
-    public User saveUser(@RequestBody User usuario){
-        return userService.saveUser(usuario);
+    public ResponseEntity<UserRespostaDTO> saveUser(@RequestBody UserDTO dto){
+      User usuario = userService.saveUser(dto.transformaParaObjeto());
+      return new ResponseEntity<>(UserRespostaDTO.transformaEmDTO(usuario), HttpStatus.CREATED);
     }
 
-    //Deletar
+    //Deletar usuário pelo ID
     @DeleteMapping("{id}")
-    @ResponseStatus(HttpStatus.OK)
-    public void deleteUserByID(@PathVariable Integer id){
-        userService.deleteUserByID(id);
+    public ResponseEntity deleteUserByID(@PathVariable Integer id){
+      userService.deleteUserByID(id);
+      return new ResponseEntity("Usuário deletado", HttpStatus.OK);
     }
 
-    //Atualizar
+    //Atualizar usuário pelo ID
     @PatchMapping("{id}")
-    @ResponseStatus(HttpStatus.OK)
-    public User updateUserByID(@RequestBody User usuario, @PathVariable Integer id){
-        if(userService.existsUserByID(id)){
-            usuario.setId(id);
-            return userService.saveUser(usuario);
-        }else{
-            return new User();
-        }
+    public ResponseEntity updateUser(@RequestBody UserDTO dto, @PathVariable Integer id){
+      if(userService.existsUserByID(id)){
+        User user = userService.findUserByID(id);
+        user.setName(dto.getName());
+        user.setEmail(dto.getEmail());
+        user.setPassword(dto.getPassword());
```

Comment:
Can you take a look at the Object Mapper pattern for Java? That way, we don't have to build objects by hand like so

Here's [an example](https://github.com/guilhXavier/openfut/tree/main/src/main/java/br/com/gtx/openfut/mapper)